examples/state_machine/events.py

```diff
@@ -0,0 +1,98 @@
+from pubnub.exceptions import PubNubException
+
+
+class PNEvent:
+    def get_name(self) -> str:
+        return self.__class__.__name__
+
+
+class PNFailureEvent(PNEvent):
+    def __init__(self, reason: PubNubException, attempt: int) -> None:
+        self.reason = reason
+        self.attempt = attempt if attempt else 0
```

Comment:
I don't think it's necessary to be here. I think the flow of the attempts information is that it always flow from the states and never into. If you remove it from here you might need to fix some lines where you take attempt from the event

Reply:
ok I moved the attempt reset to proper state constructors should be fine now